inspectit-oce-core/src/test/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContextUnitTest.java

```diff
@@ -0,0 +1,748 @@
+            doReturn(true).when(propagation).isTag(any());
+
+            InspectitContext ctx = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctx.makeActive(true);
+
+            assertThat(getCurrentTagsAsMap()).hasSize(2);
+            assertThat(getCurrentTagsAsMap()).containsEntry("tagA", "valueA");
+            assertThat(getCurrentTagsAsMap()).containsEntry("tagB", "valueB");
+
+            ctx.close();
+            assertThat(InspectitContext.INSPECTIT_KEY.get()).isNull();
+        }
+
+
+        @Test
+        void verifyLocalValuesPublishedCorrectly() {
+            when(commonTags.getCommonTagValueMap()).thenReturn(Collections.emptyMap());
+            doReturn(false).when(propagation).isPropagatedDownWithinJVM(eq("local"));
+            doReturn(true).when(propagation).isPropagatedDownWithinJVM(eq("global"));
+            doReturn(true).when(propagation).isTag(any());
+
+            InspectitContext ctx = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctx.setData("local", "localValue");
+            ctx.setData("global", "globalValue");
+
+            ctx.makeActive(true);
+
+            assertThat(getCurrentTagsAsMap()).hasSize(1);
+            assertThat(getCurrentTagsAsMap()).containsEntry("global", "globalValue");
+
+            try (Scope scope = ctx.enterNewTagScope()) {
+                assertThat(getCurrentTagsAsMap()).hasSize(2);
+                assertThat(getCurrentTagsAsMap()).containsEntry("local", "localValue");
+                assertThat(getCurrentTagsAsMap()).containsEntry("global", "globalValue");
+            }
+
+            ctx.close();
+            assertThat(InspectitContext.INSPECTIT_KEY.get()).isNull();
+        }
+
+
+        @Test
+        void verifyUpPropagatedValuesNotAvailable() {
+            when(commonTags.getCommonTagValueMap()).thenReturn(Collections.emptyMap());
+            doReturn(true).when(propagation).isPropagatedDownWithinJVM(any());
+            doReturn(true).when(propagation).isPropagatedUpWithinJVM(any());
+            doReturn(true).when(propagation).isTag(any());
+
+            InspectitContext ctx = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctx.setData("rootKey1", "rootValue1");
+            ctx.setData("rootKey2", "rootValue2");
+
+            ctx.makeActive(true);
+
+            assertThat(getCurrentTagsAsMap()).hasSize(2);
+            assertThat(getCurrentTagsAsMap()).containsEntry("rootKey1", "rootValue1");
+            assertThat(getCurrentTagsAsMap()).containsEntry("rootKey2", "rootValue2");
+
+            InspectitContext nested = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            nested.makeActive(true);
+            nested.setData("rootKey1", "nestedValue1");
+            nested.setData("nestedKey2", "nestedValue2");
+            nested.close();
+
+            assertThat(getCurrentTagsAsMap()).hasSize(2);
```

Comment:
what the heck how is this now working with no propagation?

Reply:
I will add a verification that up-propagated values are available in the full scope.